pandas/core/generic.py

```diff
@@ -7381,7 +7381,13 @@ def _clip_with_one_bound(self, threshold, method, axis, inplace):
         # GH 40420
         # Treat missing thresholds as no bounds, not clipping the values
         if is_list_like(threshold):
-            fill_value = np.inf if method.__name__ == "le" else -np.inf
+            method_name_le = method.__name__ == "le"
+            if is_datetime64_any_dtype(self.dtype):
+                fill_value = Timestamp.max if method_name_le else Timestamp.min
+            elif is_extension_array_dtype(self.dtype):
+                fill_value = self.max() if method_name_le else self.min()
```

Comment:
Is there a better way to do this for ea ints?

Reply:
`pandas/core/dtypes/missing/na_value_for_dtype`